### PR TITLE
Initial code + README commit

### PR DESCRIPTION
**Pull request by cat-haines**

### CODE REVIEW

File: LIS3DH.class.nut [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    }
+
+    // Read data from the Accelerometer
+    // Returns a table {x: <data>, y: <data>, z: <data>}
+    function getAccel(cb = null) {
+        local x_raw = (_getReg(OUT_X_H) << 8) + _getReg(OUT_X_L);
+        local y_raw = (_getReg(OUT_Y_H) << 8) + _getReg(OUT_Y_L);
+        local z_raw = (_getReg(OUT_Z_H) << 8) + _getReg(OUT_Z_L);
+
+        //server.log(format("%02X, %02X, %02X",x_raw, y_raw, z_raw));
```
Comment (DoronShapiro) [inline LIS3DH.class.nut]: Commented debug statement

File: LIS3DH.class.nut [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    // Returns the deviceID (should be 51)
+    function getDeviceId() {
+        return _getReg(WHO_AM_I);
+    }
+
+    //-------------------- INTERRUPTS --------------------//
+
+    // Enable/disable and configure inertial interrupts
+    function configureInertialInterrupt(state, threshold = 1, duration = 5, options = null) {
+        // Set default value for options (using statics, so can't set in )
```
Comment (DoronShapiro) [inline LIS3DH.class.nut]: Missing a word here?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+accel.configureClickInt(true, LIS3DH.SINGLE_CLICK);
+```
+
+#### Double Click Example
+```squirrel
+// configure a double click interrupt
+accel.configureClickInt(true, LIS3DH.DOUBLE_CLICK);
+```
+
+### configureDataReadyInterrupt(*state*)
```
Comment (DoronShapiro) [inline README.md:167]: This a "configure" method, while `setInterruptLatching()` is a "set" method, but they both just enable/disable a feature.  We should probably try to be consistent with naming here.

File: LIS3DH.class.nut [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+        value = ~(value & mask) + 1;
+        return value & mask;
+    }
+
+    function _getReg(reg) {
+        local val = _i2c.read(_addr, format("%c", reg), 1);
+        if (val != null) {
+            return val[0];
+        } else {
+            return null;
```
Comment (DoronShapiro) [inline LIS3DH.class.nut]: A `null` return type is never checked for when this method is used and may cause issues.

File: LIS3DH.class.nut [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+        }
+    }
+
+    function _setReg(reg, val) {
+        _i2c.write(_addr, format("%c%c", reg, (val & 0xff)));
```
Comment (DoronShapiro) [inline LIS3DH.class.nut]: Should we validate that I2C operations succeed?
(i.e. by checking the return value)

File: LIS3DH.class.nut [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+    }
+
+    function _setRegBit(reg, bit, state) {
```
Comment (DoronShapiro) [inline LIS3DH.class.nut:340]: What does _state_ do?

Comment (cat-haines) [inline LIS3DH.class.nut:340]: _setRegBit sets a specific bit in the register (so state is the state we're setting of that bit).

**Pull request merged**